@bmagyar @sloretz I have permissions, but Travis said this breaks the build. Can we add `if ACTIONLIB_Version >= 1.12.0` or `if not defined` to keep backcompatibility?
https://travis-ci.com/PR2/pr2_controllers/jobs/223258204
```
[ 88%] Building CXX object CMakeFiles/robot_mechanism_controllers.dir/src/joint_trajectory_action_controller.cpp.o
In file included from /tmp/ws/src/pr2_controllers/robot_mechanism_controllers/src/joint_trajectory_action_controller.cpp:37:0:
/tmp/ws/src/pr2_controllers/robot_mechanism_controllers/include/robot_mechanism_controllers/joint_trajectory_action_controller.h:81:3: error: ‘ResultPtr’ does not name a type; did you mean ‘Result’?
   ResultPtr preallocated_result_;  // Preallocated so it can be used in realtime
   ^~~~~~~~~
   Result
/tmp/ws/src/pr2_controllers/robot_mechanism_controllers/include/robot_mechanism_controllers/joint_trajectory_action_controller.h:82:3: error: ‘FeedbackPtr’ does not name a type; did you mean ‘Feedback’?
   FeedbackPtr preallocated_feedback_;
   ^~~~~~~~~~~
   Feedback
/tmp/ws/src/pr2_controllers/robot_mechanism_controllers/include/robot_mechanism_controllers/joint_trajectory_action_controller.h:84:43: error: ‘ResultPtr’ does not name a type; did you mean ‘Result’?
```

--- a/robot_mechanism_controllers/CMakeLists.txt
+++ b/robot_mechanism_controllers/CMakeLists.txt
@@ -23,7 +23,11 @@ catkin_package(
    LIBRARIES robot_mechanism_controllers
 )
 
-add_definitions(-O3)
+string(REPLACE "." ";" VERSION_LIST ${actionlib_VERSION})
+list(GET VERSION_LIST 0 actionlib_VERSION_MAJOR)
+list(GET VERSION_LIST 1 actionlib_VERSION_MINOR)
+list(GET VERSION_LIST 2 actionlib_VERSION_PATCH)
+add_definitions(-O3 -Dactionlib_VERSION=${actionlib_VERSION} -Dactionlib_VERSION_MAJOR=${actionlib_VERSION_MAJOR} -Dactionlib_VERSION_MINOR=${actionlib_VERSION_MINOR} -Dactionlib_VERSION_PATCH=${actionlib_VERSION_PATCH})
 
 add_library(robot_mechanism_controllers
     src/cartesian_pose_controller.cpp

--- a/robot_mechanism_controllers/include/robot_mechanism_controllers/joint_trajectory_action_controller.h
+++ b/robot_mechanism_controllers/include/robot_mechanism_controllers/joint_trajectory_action_controller.h
@@ -69,6 +69,10 @@ private:
 
   //typedef actionlib::ActionServer<Action>::GoalHandle GoalHandle;
   typedef actionlib::ServerGoalHandle<Action> GoalHandle;
+#if ((actionlib_VERSION_MAJOR == 1) && (actionlib_VERSION_MINOR < 12)) || (actionlib_VERSION_MAJOR < 1)
+  typedef boost::shared_ptr<Result> ResultPtr;
+  typedef boost::shared_ptr<Feedback> FeedbackPtr;
+#endif
 
   uint8_t state_;
 

